File: Easy/Second Largest/second-largest.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>

using namespace std;

// } Driver Code Ends
//User function template for C++
class Solution{
public:	
	// Function returns the second
	// largest elements
	int print2largest(int arr[], int n) {
	    // code here
	  int  large = arr[0];
	   int  slarge = -1;
	    for(int i=1;i<n;i++)
	    {
	      if(arr[i]>large)
	      {
	          slarge=large;
	          large=arr[i];
	      }
	      else if(arr[i]<large && slarge<arr[i])
	      {
	          slarge=arr[i];
	      }
	    }
	    return slarge;
	}
};

//{ Driver Code Starts.

int main() {
    int t;
    cin >> t;
    while (t--) {
        int n;
        cin >> n;
        int arr[n];
        for (int i = 0; i < n; i++) {
            cin >> arr[i];
        }
        Solution ob;
        auto ans = ob.print2largest(arr, n);
        cout << ans << "\n";
    }
    return 0;
}

// } Driver Code Ends

File: Difficulty: Easy/Second Largest/second-largest.java
//{ Driver Code Starts
// Initial Template for Java
import java.io.*;
import java.util.*;
import java.util.stream.Collectors;

public class Main {

    public static void main(String[] args) throws Exception {
        Scanner sc = new Scanner(System.in);
        int t = Integer.parseInt(sc.nextLine());
        while (t-- > 0) {
            List<Integer> arr = new ArrayList<>();
            String input = sc.nextLine();
            Scanner ss = new Scanner(input);
            while (ss.hasNextInt()) {
                arr.add(ss.nextInt());
            }
            Solution ob = new Solution();
            int ans = ob.print2largest(arr);
            System.out.println(ans);
        }
    }
}

// } Driver Code Ends


// User function Template for Java

class Solution {
    public int print2largest(List<Integer> arr) {
        // Code Here
        int n=arr.size();
        if(n<2){
            return -1;
        }
        int max=0;
        int smax=0;
        for(int i=0;i<n;i++)
            if(max<arr.get(i)){
                smax=max;
                max=arr.get(i);
            }
        else
        {
        if(arr.get(i)<max && smax<arr.get(i)){
            smax=arr.get(i);
        }
    }
        return smax;
        
    }
}
